app/controllers/global_user_roles_controller.rb

```diff
@@ -0,0 +1,75 @@
+class GlobalUserRolesController < GlobalSettingsController
+
+  before_filter :load_user, only: [:destroy, :edit, :update]
+
+  def index
+    @users = UserPresenter.wrap(User.with_global_roles)
+  end
+
+  def destroy
+    destroy_all_global_roles
+    redirect_to global_user_roles_url
+  end
+
+  def edit
+    if @user == current_user
+      flash[:error] = translate("self_not_allowed", action: "change")
+      redirect_to global_user_roles_url
+    else
+      @user = UserPresenter.new(@user)
+    end
+  end
+
+  def search
+  end
+
+  def update
+    case
+    when params[:roles].blank?
```

Comment:
The `blank?` case is there because updating while selecting no global roles is equivalent to destroying all global roles. It's mainly there so the user will see a more appropriate `flash` afterward. That is, if I deselect all global user roles and save successfully, I will see the `destroy.success` message instead of `update.success`.

`roles_from_params` was an accidental commit of something I had in a WIP. I'm removing it.